src/library/modules/BattlePrediction/rank/Rank.js

```diff
@@ -0,0 +1,155 @@
+(function () {
+  /*--------------------------------------------------------*/
+  /* --------------------[ PUBLIC API ]-------------------- */
+  /*--------------------------------------------------------*/
+
+  const parseStartJson = (battleData) => {
+    const { makeShips, removeRetreated } = KC3BattlePrediction.rank;
+
+    const main = makeShips(battleData.api_nowhps, battleData.api_maxhps);
+    const escort = makeShips(battleData.api_nowhps_combined, battleData.api_maxhps_combined);
+
+    return {
+      playerMain: removeRetreated(main.player, battleData.api_escape_idx),
+      playerEscort: removeRetreated(escort.player, battleData.api_escape_idx_combined),
+      enemyMain: main.enemy,
+      enemyEscort: escort.enemy,
+    };
+  };
+
+  const normalizeFleets = (battleResult) => {
+    const { omitEmptySlots, hideOverkill } = KC3BattlePrediction.rank;
+
+    return Object.keys(battleResult).reduce((result, key) => {
+      return Object.assign(result, {
+        [key]: hideOverkill(omitEmptySlots(battleResult[key])),
+      });
+    }, {});
+  };
+
+  const predictors = {
+    ld_airbattle: 'airRaid',
+  };
```

Comment:
oh, should not be full string matched here... the `api_name` is not official api property, it's added in our `Kcsapi.js` (shorten ver of request url), that's why uses `indexOf` matching.